### PR TITLE
Fix log10 warning in step_detached.py

### LINKED ISSUE #635: [BUG]: RuntimeWarning: invalid value encountered in `step_detached`
**Describe the bug**
In some initial-final population runs, `step_detached` is throwing warnings. I see this with many binaries. 
```
/home/users/b/briel/software_versions/development/POSYDON/posydon/binary_evol/DT/step_detached.py:1688: RuntimeWarning: invalid value encountered in log10
  current = np.log10(
```

This is part of this but of code:
https://github.com/POSYDON-code/POSYDON/blob/7ae985fcba31489f248f1e49a97056de68143896/posydon/binary_evol/DT/step_detached.py#L1683-L1695

The solver output might be similar to the input values?

**Expected behavior**
Make sure no invalid value is inputted into the log10.

**POSYDON Version:**
POSYDON v2.0.0

**Additional context**

### PR DESCRIPTION
**Pull request by YuvanRajMathan**

### CODE REVIEW

**Comment by sgossage** [comment]
Thank you for this fix @YuvanRajMathan! Please add a short description to this PR summarizing the changes.

**Comment by YuvanRajMathan** [comment]
> Thank you for this fix @YuvanRajMathan! Please add a short description to this PR summarizing the changes.

This PR fixes the RuntimeWarning that occurs in `step_detached.py` when calculating `log_total_angular_momentum`. The warning appears when trying to take the logarithm of zero or negative values.

Changes made:
- Added safety checks before calculating logarithms
- Handle zero/negative angular momentum cases by returning -98.99 (consistent with other parts of the codebase)
- Added test case to verify the fix

The fix prevents the warning:
```RuntimeWarning: invalid value encountered in log10```

This makes the code more robust by properly handling edge cases in angular momentum calculations.
```

5. After filling in the title and description:
   - Review the changes shown in the "Files changed" tab
   - Click "Create pull request"

Would you like me to explain any part of this process in more detail? Or would you like help with creating the pull request?

File: posydon/binary_evol/DT/step_detached.py [hunk 1]
```diff
@@ -1685,14 +1685,20 @@ def get_star_profile(star, htrack, m0):
...
-                            history = np.log10(
-                                (interp1d_pri["omega"][:-1] / const.secyer)
-                                * (interp1d_pri[self.translate["total_moment_of_inertia"]](
-                                        t[:-1] - t_offset_pri) * (const.msol * const.rsol**2)))
+                            # Calculate the angular momentum
+                            ang_momentum = (interp1d_pri["omega"][-1] / const.secyer) * (
+                                interp1d_pri[self.translate["total_moment_of_inertia"]](
+                                    t[-1] - t_offset_pri).item() * (const.msol * const.rsol**2))
+                            # Handle zero or negative values
+                            current = np.log10(ang_momentum) if ang_momentum > 0 else -98.99
```
Comment (sgossage) [inline posydon/binary_evol/DT/step_detached.py:1693]:
```suggestion
                            current = np.log10(ang_momentum) if ang_momentum > 0 else -99
```

To conform to other areas of the code where we use -99 or 99 as a value in such cases, I suggest making this change.

File: posydon/binary_evol/DT/step_detached.py [hunk 2]
```diff
@@ -1685,14 +1685,20 @@ def get_star_profile(star, htrack, m0):
...
+                            
+                            # Calculate history values with the same safety check
+                            history_momentum = (interp1d_pri["omega"][:-1] / const.secyer) * (
+                                interp1d_pri[self.translate["total_moment_of_inertia"]](
+                                    t[:-1] - t_offset_pri) * (const.msol * const.rsol**2))
+                            history = np.where(history_momentum > 0, 
+                                             np.log10(history_momentum), 
+                                             -98.99)
```
Comment (sgossage) [inline posydon/binary_evol/DT/step_detached.py:1701]:
```suggestion
                                             -99)
```

**Comment by sgossage** [comment]
Note that this code would be overwritten by the rewrite of the detached step ([PR#511](https://github.com/POSYDON-code/POSYDON/pull/511)). I've incorporated the suggested changes of this PR within PR#511 to preserve them.

We should either accept this PR before merging PR#511, or simply merge PR#511 and close this PR.

**Comment by astroJeff** [comment]
These changes have been integrated upstream into PR #511. Closing.
